src/pages/videolist/VideoPage.jsx

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from "react";
+import { CategoryFilter, Header, Navpills, SideNav } from "../../components";
+import { MdMoreVert } from "react-icons/md";
+import axios from "axios";
+import "./videolist.css";
+import "./videopage.css";
+import { useFilter } from "../../contexts/filter-context/filter-context";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import ReactPlayer from "react-player";
+
+export function VideoPage() {
+  const [video, setVideo] = useState({});
+  const { id } = useParams();
+
+  async function getVideoDetails() {
+    try {
+      const { data } = await axios.get(`/api/video/${id}`);
+      setVideo(() => data.video);
+    } catch (e) {
+      console.log("error", e);
+    }
+  }
+  useEffect(() => getVideoDetails(), []);
+  return (
+    <>
+      <Header />
+      <SideNav />
```

Comment:
This is not react way of doing things. You define the common components once in the file where you are defining routes

```JSX
<BrowserRouter>
     <Header />
     <SIdebar />
     <Routes>
           <Route  />
           <Route />
     <Routes />
</BrowserRouter>
```

This way the Header and sidebar are always rendered and the route matching with the address in address bar will be displayed along with them

Reply:
Oh i did not know this, also I might not need these in some components so in that case what is the better way?